src/Resolver.php

```diff
@@ -541,6 +541,7 @@ public function requireAndResolveColumns(array $columns, Model $model = null)
     {
         $model = $model ?: $this->query->getModel();
         $tableName = $model->getTableName();
+        $targetColumns = $model->getColumns();
```

Comment:
That's not correct. `$model` is not necessarily the final `$target`, from which the columns should be fetched instead once it's available. (after the switch-case)